### PR TITLE
Issue 58 Add wakelock and brightness handler

### PR DESCRIPTION
**Pull request by miika22**

### CODE REVIEW

**Comment by miika22** [comment]
[Create brightness handler](https://trello.com/c/dGXyMrbl/58-create-brightness-handler)

**Comment by miika22** [comment]
These features are done. InactivityHandler now dims after 15 seconds of inactivity (only on HomeScreen). It also returns to HomeScreen from anywhere if AppState goes from "background" to "active" (uses navigation's popToTop).

File: src/components/InactivityHandler.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            this.onActiveChange(false);
+        }, this.props.idleTime);
+    }
+
+    /**
+     * Called to handle active change.
+     * If active, set brightness to system brightness.
+     * If inactive, dim brightness to 0.
+     */
+    public onActiveChange(active: boolean): void {
```
Comment (ChaoticWeg) [inline src/components/InactivityHandler.tsx:135]: After click-through testing, I found that in iOS it looks like Expo's `Brightness.setBrightnessAsync(...)` directly modifies the system brightness. I'm not sure if it does the same on Android, but the effect is that:

- Dimming on inactivity directly modifies the system brightness
- The subsequent call to `Brightness.getSystemBrightnessAsync()` returns the already-dimmed brightness, meaning that...
- Returning the brightness to current system brightness on activity has no effect

My proposed solution is to grab the system brightness in `componentWillMount()` and hold onto it as the "previous" or "normal" system brightness.

**Review by ChaoticWeg — CHANGES_REQUESTED** [review]
Everything looks good! Just this functionality issue that should be straightforward. Might be hard to test if I'm not around, unless @richardkriesman can bust out his iOS VD and give it a once-over.

File: src/components/InactivityHandler.tsx (hunk 1 above)
Comment (miika22) [reply src/components/InactivityHandler.tsx:135]: Yeah, it seems like it works a bit differently in iOS. I like the proposed solution, will add that later today

**Review by ChaoticWeg — APPROVED** [review]
With that change, the wakelock and dimming works beautifully on iOS! Approving.

**Pull request merged**